src/Spice86.Core/Emulator/VM/Breakpoint/Serializable/EmulatorBreakpointsSerializer.cs

```diff
@@ -0,0 +1,115 @@
+﻿namespace Spice86.Core.Emulator.VM.Breakpoint.Serializable;
+
+using Spice86.Shared.Interfaces;
+using Spice86.Shared.Emulator.VM.Breakpoint.Serializable;
+using Spice86.Core.CLI;
+using Spice86.Shared.Utils;
+
+using System;
+using System.Text.Json;
+using System.Text;
+using System.IO;
+using System.Security.Cryptography;
+
+/// <summary>
+/// Serializes (on startup) and deserialize (on app shutdown) internal debugger breakpoints from disk.
+/// </summary>
+public class EmulatorBreakpointsSerializer {
+    private const string BreakpointsFileNameFormat = "Breakpoints_{0}.json";
+    private readonly Configuration _configuration;
+    private readonly ILoggerService _loggerService;
+    private ISerializableBreakpointsHolder? _serializableBreakpointsHolder = null;
+    private readonly string _programHash;
+
+    public EmulatorBreakpointsSerializer(Configuration configuration, ILoggerService loggerService) {
+        _configuration = configuration;
+        _loggerService = loggerService;
+        _programHash = GetProgramHash(configuration);
+    }
+
+    private string GetProgramHash(Configuration configuration) {
```

Comment:
The configuration value is just there for one purpose which is to restrict execution in case hash don't match. This is mainly used for overrides.

I suggest you make spice86 compute the exe hash all the time (in a separate class) and use that to:
- Check if param matches actual exe hash (current use case)
- Generate the folder name (this use case)

Generation of folder name shouldn't be there though, it should be done in case there is no configuration given to specify the dump directory (RecordedDataDirectory parameter)

Reply:
There's no folder generation, we simply pass the dirPath from EmulatorStateSerializer now.

The SHA256 hash is always created, as suggested.